recipes/jemalloc/all/conanfile.py

```diff
@@ -81,20 +84,21 @@ def validate(self):
             raise ConanInvalidConfiguration("Only Release and Debug build_types are supported")
         if self.settings.compiler == "Visual Studio" and self.settings.arch not in ("x86_64", "x86"):
             raise ConanInvalidConfiguration("Unsupported arch")
-        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) <= "3.9":
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) <= "3.9":
             raise ConanInvalidConfiguration("Unsupported compiler version")
+        if self.settings.os == "Macos" and self.settings.arch not in ("x86_64", "x86"):
+            raise ConanInvalidConfiguration("Unsupported arch")
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
             self.build_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
```

Comment:
It's weird to see `self.source_folder` without layout 😳
Regardless  https://github.com/conan-io/conan-center-index/blob/dafd47d3f5cf02ddd0c6d425f229faf01b2fee9b/recipes/jemalloc/all/conanfile.py#L56 is unused

Reply:
I understood you want this method to be restored:

https://github.com/conan-io/conan-center-index/pull/12859#discussion_r979372375

Did I misunderstand?

Reply:
I appogize, I've been traveling so responses are not the best 🙊 

I meant removing this method all together, no need for comments... sorry for the confusion